src/fu-self-test.c

```diff
@@ -6688,6 +6688,12 @@ fu_test_engine_fake_serio(gconstpointer user_data)
 	g_assert_no_error(error);
 	g_assert_true(ret);
 
+	/* no gnutls */
```

Comment:
I think you should be using `#ifndef HAVE_GNUTLS` to wrap this

Reply:
I originally had this; but I figured it would be better to not duplicate the logic we used in `plugins/synaptics-rmi/meson.build` -- i.e. we'd have to check the value of `option('plugin_synaptics_rmi')` too.

Reply:
Well wait a minute - how `plugin_synaptics_rmi` even being set?  It's supposed to be excluded when built without gnutls.

I just double checked.  When I do `-Dgnutls=disabled` I get:

```
  plugins
    enabled             : acpi-dmar, acpi-facp, acpi-ivrs, acpi-phat, algoltek-aux, algoltek-usb, algoltek-usbcr, amd-gpu, amd-kria, amd-pmc, analogix, android-boot, ata, aver-hid, bcm57xx, bios, ccgx, ccgx-dmc, cfu, ch341a, ch347, colorhug, corsair, cpu, cros-ec, dell, dell-dock, dell-k2, dfu, dfu-csr, ebitdo, elanfp, elantp, emmc, ep963x, fastboot, flashrom, focalfp, fpc, fresco-pd, genesys, genesys-gl32xx, goodix-moc, goodix-tp, gpio, hailuck, hpi-cfu, huddly-usb, intel-gsc, intel-me, intel-usb4, iommu, jabra, jabra-file, jabra-gnp, kinetic-dp, legion-hid2, lenovo-thinklmi, linux-display, linux-lockdown, linux-sleep, linux-swap, linux-tainted, logind, logitech-bulkcontroller, logitech-hidpp, logitech-rallysystem, logitech-scribe, logitech-tap, mediatek-scaler, modem-manager, msr, mtd, nitrokey, nordic-hid, nvme, optionrom, parade-lspcon, parade-usbhub, pci-bcr, pci-mei, pci-psp, pixart-rf, powerd, qc-s5gen2, qsi-dock, realtek-mst, redfish, rts54hid, rts54hub, scsi, steelseries, synaptics-cape, synaptics-cxaudio, synaptics-mst, synaptics-prometheus, synaptics-vmm9, system76-launch, telink-dfu, test, thelio-io, thunderbolt, ti-tps6598x, tpm, uefi-capsule, uefi-dbx, uefi-esrt, uefi-recovery, uefi-sbat, uf2, upower, usi-dock, vbe, vli, wacom-raw, wacom-usb, wistron-dock
    disabled            : synaptics-rmi, uefi-pk
```

Reply:
I think I follow it now; the daemon does have no synaptics-rmi, so you **expect** such an error and you don't reinvent the logic.

OK, this makes sense, sorry for the confusion.